state-chain/custom-rpc/src/lib.rs

```diff
@@ -781,6 +781,18 @@ pub trait CustomApi {
 		additional_orders: Option<Vec<SwapRateV2AdditionalOrder>>,
 		at: Option<state_chain_runtime::Hash>,
 	) -> RpcResult<RpcSwapOutputV2>;
+	#[method(name = "swap_rate_v4")]
+	fn cf_pool_swap_rate_v4(
+		&self,
+		from_asset: Asset,
+		to_asset: Asset,
+		amount: U256,
+		broker_commission: BasisPoints,
+		dca_parameters: Option<DcaParameters>,
+		ccm_data: Option<CcmData>,
```

Comment:
> we can consider this a non-breaking change and just update the v3 rpc
But it will break anyone using v3 if we update it like you suggest, no?

v2 and v3 were added and just had one or two new fields, that's why I went for v4. I don't have a strong opinion there, happy to just add it to v3 if you think that's better.

Reply:
v2 changed the output type in a breaking way.
v3 added a new *required* input param (breaking change)
v4 adds a new optional param -> this is not a breaking change, so we can just add it to the v3 rpc.

Reply:
Oh right - I've just been too long in the blockchain encoding trenches where everything in the encoding is a breaking change :facepalm: but of course this isn't. Will fix.